temporal-sdk/src/main/java/io/temporal/internal/sync/SyncWorkflowContext.java

```diff
@@ -136,6 +137,15 @@ public void setHeadInterceptor(WorkflowOutboundCallsInterceptor head) {
     }
   }
 
+  @Override
+  public void setDefaultActivityOptions(ActivityOptions defaultActivityOptions) {
+    this.defaultActivityOptions = defaultActivityOptions;
+  }
+
+  public ActivityOptions getDefaultActivityOptions() {
+    return this.defaultActivityOptions;
+  }
```

Comment:
I would also add public Workflow.setDefaultActivityOptions and Workflow.setActivityOptions.